src/gtest/Makefile

```diff
@@ -0,0 +1,146 @@
+CXX ?= g++
+OPT=-O3 -DNDEBUG
+
+ifeq ($(DEBUG_BUILD),1)
+	CXX+=-DDEBUG_BUILD=1
+endif
+
+.DEFAULT_GOAL := all
+SOURCES := $(wildcard *.cpp)
+ALL_OBJECTS := $(patsubst %.cpp,%.o,$(SOURCES))
+DEPENDS := $(patsubst %.cpp,%.d,$(SOURCES))
+
+MORE_WARNINGS=-Wold-style-cast \
+	-Wno-variadic-macros \
+	-Werror \
+	-Wpedantic \
+	-Wextra \
+	-Wall \
+	-Wcast-align \
+	-Wcast-qual \
+	-Wframe-larger-than=32768 \
+	-Wno-strict-overflow \
+	-Wsync-nand \
+	-Wtrampolines \
+	-Wsign-compare \
+	-Werror=float-equal \
+	-Werror=missing-braces \
+	-Werror=init-self \
+	-Werror=logical-op \
+	-Werror=write-strings \
+	-Werror=address \
+	-Werror=array-bounds \
+	-Werror=char-subscripts \
+	-Werror=enum-compare \
+	-Werror=empty-body \
+	-Werror=main \
+	-Werror=aggressive-loop-optimizations \
+	-Werror=nonnull \
+	-Werror=parentheses \
+	-Werror=return-type \
+	-Werror=sequence-point \
+	-Werror=uninitialized \
+	-Werror=volatile-register-var \
+	-Werror=ignored-qualifiers \
+	-Wno-unused-function \
+	-Wno-missing-field-initializers \
+	-Wdouble-promotion \
+	-Wformat=2
+
+-include $(DEPENDS)
+
+ifdef COVERAGE
+	TEST_CFLAGS += -DCOVERAGE_TEST $(COVERAGE_CFLAGS)
+endif
+
+ifdef FORCE_RUN_SKIPPED_TESTS
+	TEST_CFLAGS+=-DFORCE_RUN_SKIPPED_TESTS
+endif
+
+# Track compilation flags to force rebuild when they change
+.PHONY: .flags
+.flags:
+	@echo '$(TEST_CFLAGS)' | cmp -s - $@ || echo '$(TEST_CFLAGS)' > $@
+
+# OSS Valkey library dependencies
+VALKEY_LIB := ../libvalkey.a
+HIREDIS_LIB := ../../deps/libvalkey/lib/libvalkey.a
+LUA_LIB := ../../deps/lua/src/liblua.a
+JEMALLOC_LIB := ../../deps/jemalloc/lib/libjemalloc.a
+HDR_HISTOGRAM_LIB := ../../deps/hdr_histogram/libhdrhistogram.a
+FPCONV_LIB := ../../deps/fpconv/libfpconv.a
+GTEST_LIB := ../../deps/googletest/build/googlemock/gtest/libgtest.a
+GMOCK_LIB := ../../deps/googletest/build/googlemock/libgmock.a
+
+# Ensure googletest is built
```

Comment:
In the cases where we want to say `GoogleTest` in text/comments, let's make sure the capitalization is correct.